front/src/applications/editor/tools/routeEdition/tool.tsx

```diff
@@ -0,0 +1,120 @@
+import { RouteEditionState } from './types';
+import { Layers, LeftPanel, Messages } from './components';
+import { getEmptyRoute, getRouteEditionState } from './utils';
+
+const RouteEditionTool: Tool<RouteEditionState> = {
+  id: 'route-edition',
+  icon: GiPathDistance,
+  labelTranslationKey: 'Editor.tools.routes-edition.label',
+  requiredLayers: new Set(['buffer_stops', 'detectors']),
+  getInitialState: () => getRouteEditionState(),
+  actions: [
+    [
+      {
+        id: 'save-route',
+        icon: AiFillSave,
+        labelTranslationKey: 'Editor.tools.routes-edition.actions.save-route',
+        isDisabled({ state: { isComplete } }) {
+          return !isComplete;
+        },
+        async onClick({ setIsFormSubmited }) {
+          if (setIsFormSubmited) {
+            setIsFormSubmited(true);
+          }
+        },
+      },
+      {
+        id: 'reset-route',
+        icon: BiReset,
+        labelTranslationKey: `Editor.tools.routes-edition.actions.reset-route`,
+        isDisabled({ state: { entity, initialEntity } }) {
+          return isEqual(entity, initialEntity) || isEqual(entity, getEmptyRoute());
+        },
+        onClick({ state: { initialEntity }, setState }) {
+          setState(getRouteEditionState(initialEntity));
+        },
+      },
+    ],
+    [
+      {
+        id: 'new-route',
+        icon: GoPlusCircle,
+        labelTranslationKey: 'Editor.tools.routes-edition.actions.new-route',
+        onClick({ setState }) {
+          setState(getRouteEditionState());
+        },
+      },
+    ],
+    [
+      {
+        id: 'delete-route',
+        icon: GoTrash,
+        labelTranslationKey: 'Editor.tools.routes-edition.actions.delete-route',
+        // Show button only if we are editing
+        isDisabled({ state: { entity } }) {
+          return !entity?.properties.id;
+        },
+        onClick({ infraID, openModal, closeModal, forceRender, state, setState, dispatch, t }) {
+          openModal(
+            <ConfirmModal
+              title={t('Editor.tools.routes-edition.delete-route')}
+              onConfirm={async () => {
+                if (state.entity) {
+                  await dispatch<ReturnType<typeof save>>(
+                    save(infraID, { delete: [state.entity] })
+                  );
+                  setState(getRouteEditionState());
+                  closeModal();
+                  forceRender();
+                }
+              }}
+            >
+              <p>{t('Editor.tools.routes-edition.delete-route-success').toString()}</p>
+            </ConfirmModal>
+          );
+        },
+      },
+    ],
+  ],
+
+  getCursor({ state }) {
+    if (state.extremityState.type === 'selection' && state.hovered) return 'pointer';
+    return 'default';
+  },
+  onKeyDown(e, { state, setState }) {
+    if (state.extremityState.type === 'selection' && e.key === 'Escape') {
+      setState({
+        ...state,
+        extremityState: { type: 'idle' },
+      });
+    }
```

Comment:
The short key doesn't seem to work

Reply:
For this one, it's a good cacth !
The `onKeyPress` is registered on the map, not on the window ... 
So if you click somewhere on the map and then press`escape` it works.